addon/components/au-icon.gts

```diff
@@ -6,7 +6,7 @@ export interface AuIconSignature {
     alignment?: 'left' | 'right';
     // TODO: We should deprecate the non-boolean versions since there is no reason to support them
     ariaHidden?: boolean | 'true' | 'false';
-    icon: string;
+    icon: string | Component<Record<string, never>>;
```

Comment:
I think you want to use the `ComponentLike` type here: https://typed-ember.gitbook.io/glint/using-glint/glint-types#componentlike-helperlike-and-modifierlike

Reply:
Ah, didn't know that existed, thanks.